rust/worker/src/segment/record_segment.rs

```diff
@@ -628,7 +630,7 @@ pub struct RecordSegmentReader<'me> {
     user_id_to_id: BlockfileReader<'me, &'me str, u32>,
     id_to_user_id: BlockfileReader<'me, u32, &'me str>,
     id_to_data: BlockfileReader<'me, u32, DataRecord<'me>>,
-    curr_max_offset_id: Arc<AtomicU32>,
+    max_offset_id: u32,
```

Comment:
I needed to change this because the shared state led to a bug in the log materialization logic when running multiple materializations in parallel (see comment in `compact.rs`). Readers should not expose mutable state.